game/src/game/attack/mod.rs

```diff
@@ -161,37 +173,69 @@ pub fn attack_damage(
                 .with_filter(attack_collider.0.collision_groups().filter | GROUND),
             Some(entity),
         );
-        for (entity, mut health, collider, gent, is_defending) in damageable_query.iter_mut() {
-            if colliding_entities.contains(&entity)
-                && attack.damaged.iter().find(|x| x.0 == entity).is_none()
-            {
-                let damage_dealt = if is_defending {
-                    attack.damage / 4
-                } else {
-                    attack.damage
-                };
-                health.current = health.current.saturating_sub(damage_dealt);
-                attack.damaged.push((entity, time.tick(), damage_dealt));
-                if let Ok((anim_entity, mut anim_player)) = gfx_query.get_mut(gent.e_gfx) {
-                    // is there any way to check if a slot is set?
-                    anim_player.set_slot("Damage", true);
-                    commands.entity(anim_entity).insert(DamageFlash {
-                        current_ticks: 0,
-                        max_ticks: 8,
-                    });
-                }
-                if health.current == 0 {
-                    commands.entity(entity).insert(Dead);
+
+        let mut collided: Vec<usize> = Vec::new();
+        for entity in colliding_entities.iter() {
+            let mut should_apply_damage = true;
+            for (i, attack) in attack.damaged.iter_mut().enumerate() {
+                if attack.entity == *entity {
+                    collided.push(i);
+                    if attack.contact == true {
+                        should_apply_damage = false;
+                    }
                 }
-                if let Some(pushback) = maybe_pushback {
-                    commands.entity(entity).insert(Knockback::new(
-                        pushback.direction,
-                        pushback.strength,
-                        16,
-                    ));
+            }
+            if should_apply_damage {
+                if let Ok((entity, mut health, collider, gent, is_defending)) =
+                    damageable_query.get_mut(*entity)
+                {
```

Comment:
Reworked the order this is called to be a bit more efficient, this way we don't have to iterate through every single damageable entity.